dvc/repo/init.py

```diff
@@ -63,15 +63,22 @@ def init(root_dir=os.curdir, no_scm=False, force=False):
     Raises:
         KeyError: Raises an exception.
     """
+
+    if no_scm and subdir:
+        raise InitError(
+            "Cannot initialize repo with `--no-scm` and `--subdir`"
+        )
```

Comment:
OK I'm still a bit confused about the limitations here. Why can't a subdir project be no-scm?

What if the parent is no-scm? And i that's not allowed, where is the check? Thanks!

Reply:
@jorgeorpinel 
In case of initializing `--subdir`, we will make Git looks for git root inside parent directories,
so if we have this flag, and there is no parent scm, git will throw `InvalidRepositoryError`, which in result will make us initialize NoSCM, and since there is not `--no-scm` flag, it will raise InitError. 
sample script, working with current version of change:

```
#!/bin/bash

rm -rf repo
mkdir repo

pushd repo
mkdir subrepo
pushd subrepo

dvc init --subdir
```

Reply:
@pared Sorry, but I didn't get your explanation. Since `no-scm` is explicit now, `--no-scm --subdir` should be a totally valid case. Unless I'm missing something.

Reply:
@efiop well my explanation was valid when @jorgeorpinel asked the question. After our discussion on how to handle SCM, and changing the logic, it is technically possible to do that. Still, I would leave this check as using both flags at the same time indicates a misunderstanding of how they work.

Reply:
Yeah I guess it doesn't make sense to have a --subdir DVC project that does not put outputs into .gitignore if the parent DVC repo does.

Although... We do allow `dvc init --no-scm` in Git-tracked directories, so the enforced restriction is kind of inconsistent.

Same discussion as https://github.com/iterative/dvc/issues/2472#issuecomment-590184536 I guess...

Reply:
But I do agree with this restriction. I just think it should also be applied to `dvc init --no-scm` in Git-tracked dirs. See https://github.com/iterative/dvc/issues/2472#issuecomment-590184536